shell/platform/darwin/ios/framework/Source/FlutterDartProject.mm

```diff
@@ -89,14 +101,19 @@
     // In case the application bundle is still not specified, look for the App.framework in the
     // Frameworks directory.
     if (settings.application_library_path.size() == 0) {
-      NSString* applicationFrameworkPath = [mainBundle pathForResource:@"Frameworks/App.framework"
-                                                                ofType:@""];
-      if (applicationFrameworkPath.length > 0) {
-        NSString* executablePath =
-            [NSBundle bundleWithPath:applicationFrameworkPath].executablePath;
-        if (executablePath.length > 0) {
-          settings.application_library_path.push_back(executablePath.UTF8String);
-        }
+      NSString* libraryPath = FindFrameworkApplicationLibraryPath(mainBundle);
+      if (libraryPath.length > 0) {
+        settings.application_library_path.push_back(libraryPath.UTF8String);
+      }
+    }
+  }
+
+  // Check other bundles if we still haven't found anything.  This is useful for XCTest targets.
+  if (settings.application_library_path.size() == 0) {
+    for (NSBundle* bundle in [NSBundle allBundles]) {
```

Comment:
What happens if I have an add2app case here with multiple App.frameworks embedded in different bundles?  Can that happen?

Reply:
This doesn't get executed unless you don't specify the dart project, and only if you didn't find "info.plist[FLTLibraryPath]" and "mainBundle/App.framework".  I could make it more specific to only look in bundles that are named ".xctest".  I don't know if it is necessary though.

Reply:
I'm worried we're missing something here that this could break unexpectedly, but I can't think of what.  It's probably fine.